### PR TITLE
dev-lang/micropython: add 1.23.0

### PR DESCRIPTION
**Pull request by tfemby**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-lang/micropython/micropython-1.23.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs
+
+DESCRIPTION="Python implementation for microcontrollers"
+HOMEPAGE="https://github.com/micropython/micropython"
```
Comment (Jamim) [inline dev-lang/micropython/micropython-1.23.0.ebuild:9]: _Hello @tfemby,_

Could you please add the official website to the `HOMEPAGE` variable?

```suggestion
HOMEPAGE="https://micropython.org https://github.com/micropython/micropython"
```

_Thanks!_

**Review by Jamim — CHANGES_REQUESTED** [review]

File: dev-lang/micropython/micropython-1.23.0.ebuild (hunk 1 above)
Comment (tfemby) [reply dev-lang/micropython/micropython-1.23.0.ebuild]: Hey, sorry that I missed that. I added your suggestion :)

**Review by Jamim — APPROVED** [review]
Thank you for your contribution! 🙇🏼

**Review by juippis — APPROVED** [review]
Tests fail, but looks good otherwise.
```
924 tests performed (28244 individual testcases)
921 tests passed
31 tests skipped: builtin_next_arg2 builtin_range_binop cexample_class cexample_module class_inplace_op2 cmd_parsetree cryptolib_aes128_ctr deflate_compress deflate_stream_error extra_coverage ffi_types heap_locked io_buffered_writer machine_i2s_rate machine_timer machine_uart_tx memoryview_itemsize namedtuple_asdict nanbox_smallint re_debug re_groups re_span repl_sys_ps1_ps2 ssl_ioctl ssl_poll sys_getsizeof sys_settrace_features sys_settrace_generator sys_settrace_loop sys_tracebacklimit unicode_ure
3 tests failed: float_parse float_parse_doubleprec re_sub
make: *** [Makefile:259: test] Error 1
 * ERROR: dev-lang/micropython-1.23.0::gentoo failed (test phase):
 *   emake failed
```
Thanks!

**Pull request closed without merging**